spec/mailers/users_mailer_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'rails_helper'
+
+RSpec.describe UsersMailer, type: :mailer do
+  before(:each) do
+    ActionMailer::Base.delivery_method = :test
+    ActionMailer::Base.perform_deliveries = true
+    ActionMailer::Base.deliveries = []
+    @user = FactoryBot.create(:user, :sale)
+    UsersMailer.credentials(@user).deliver
+  end
+
+  after(:each) do
+    ActionMailer::Base.deliveries.clear
+  end
+
+  describe 'credentials' do
+    it 'should send an email' do
```

Comment:
do not use ActionMailer::Base.deliveries.first or anything like that

rewrite specs like here
https://relishapp.com/rspec/rspec-rails/docs/mailer-specs

but use "let!"